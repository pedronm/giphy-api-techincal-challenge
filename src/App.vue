<template>
  <div id="app">
    <header>
      <div id="nav">
        <router-link to="/search">Search</router-link> |
        <router-link to="/most-popular">Most Popular</router-link> |
        <router-link to="/random">Random</router-link>
      </div>
    </header>
    <router-view id="view" />
    <footer>
      <p>Powered by</p>
      <img alt="Giphy Icon" src="./assets/giphy-branding/GIPHY Logo 75px.png" />
    </footer>
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import store from "./store";

export default Vue.extend({
  created() {
    store.dispatch("fetchMostPopular");
  }
});
</script>

<style lang="scss">
@import "./styles/global.scss";
</style>
