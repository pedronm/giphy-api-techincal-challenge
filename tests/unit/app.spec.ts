import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import App from "@/App.vue";

const localVue = createLocalVue();
const router = new VueRouter();

localVue.use(VueRouter);

describe("App.vue", () => {
    it("renders", () => {
        const wrapper = shallowMount(App, {
            localVue,
            router
        });
        wrapper.html()
        expect(wrapper.html()).toMatchSnapshot();
    });
});