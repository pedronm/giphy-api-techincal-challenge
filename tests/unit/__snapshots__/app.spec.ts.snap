// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders 1`] = `
<div id="app">
  <header>
    <div id="nav">
      <router-link-stub to="/search" tag="a" event="click">Search</router-link-stub> |
      <router-link-stub to="/most-popular" tag="a" event="click">Most Popular</router-link-stub> |
      <router-link-stub to="/random" tag="a" event="click">Random</router-link-stub>
    </div>
  </header>
  <router-view-stub name="default" id="view"></router-view-stub>
  <footer>
    <p>Powered by</p> <img alt="Giphy Icon" src="./assets/giphy-branding/GIPHY Logo 75px.png">
  </footer>
</div>
`;
